app/models/application_state.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class ApplicationState < ApplicationRecord
+  belongs_to :application, touch: true
```

Comment:
could we not include the touch here? We're going to rethink the whole touch model :)